test_genie_python_dae.py

```diff
@@ -118,29 +144,53 @@ def test_GIVEN_running_instrument_WHEN_block_logging_THEN_block_saved_in_file(se
+            if in_alarm:
+                break
+            sleep(1)
+        self.assertTrue(in_alarm, "Block never went invalid when IOC stopped")
+
         run_number = g.get_runnumber()
         g.end()
 
         g.waitfor_runstate("SETUP", maxwaitsecs=self.TIMEOUT)
 
         nexus_path = r'/raw_data_1/selog/{}/value_log'.format(test_block_name)
-        num_to_test = 4
-        nexus_filepath = "C:/data/{instrument}{run}.nxs".format(instrument=g.adv.get_instrument(), run=run_number)
-        with h5py.File(nexus_filepath, "r") as f:
-            is_valid = [sample == 1 for sample in f[nexus_path + r'/value_valid'][-num_to_test:]]
-            value = [int(val) for val in f[nexus_path + r'/value'][-num_to_test:]]
-            severity = [str(sample[0], 'utf-8').strip() for sample in f[nexus_path + r'/alarm_severity'][-num_to_test:]]
-            alarm_status = [str(sample[0], 'utf-8').strip() for sample in f[nexus_path + r'/alarm_status'][-num_to_test:]]
-            alarm_time = [int(time) for time in f[nexus_path + r'/alarm_time'][-num_to_test:]]
-
-        self.assertListEqual(is_valid, [False, True, True, True])
-        expected_values = [0] + test_values
-        self.assertListEqual(value, expected_values)
-        self.assertListEqual(severity, ["INVALID", "NONE", "MINOR", "MAJOR"])
-        self.assertListEqual(alarm_status, ["UDF_ALARM", "NO_ALARM", "LOW_ALARM", "LOLO_ALARM"])
-
-        self.assertAlmostEqual(alarm_time[-1]-alarm_time[-2], sleep_between_sets, delta=1)
-        self.assertAlmostEqual(alarm_time[-2]-alarm_time[-3], sleep_between_sets, delta=1)
+
+        def test_function(f):
+            is_valid = [sample == 1 for sample in f[nexus_path + r'/value_valid'][:]]
+            values = [int(val) for val in f[nexus_path + r'/value'][:]]
+            severity = [str(sample[0], 'utf-8').strip() for sample in f[nexus_path + r'/alarm_severity'][:]]
+            alarm_status = [str(sample[0], 'utf-8').strip() for sample in f[nexus_path + r'/alarm_status'][:]]
+            alarm_time = [int(time) for time in f[nexus_path + r'/alarm_time'][:]]
+
+            # There could be some samples at the beginning/end but we only care about the ones we've set
+            first_value_index = values.index(10)
+            first_alarm_index = severity.index("NONE")
+
+            # Only care about test values and the final invalid one
+            is_valid = is_valid[first_value_index:first_value_index + len(test_values) + 1]
+            values = values[first_value_index:first_value_index + len(test_values) + 1]
+            severity = severity[first_alarm_index:first_alarm_index + len(test_values) + 1]
+            alarm_status = alarm_status[first_alarm_index:first_alarm_index + len(test_values) + 1]
+            alarm_time = alarm_time[first_alarm_index:first_alarm_index + len(test_values) + 1]
+
+            self.assertListEqual(is_valid, [True, True, True, False])
+            self.assertListEqual(values, test_values + [0])
```

Comment:
Could you add a comment saying why you are adding [0] - i assume it is from the value logged on end?